### PR TITLE
[KOGITO-9103] [KSW-Operator] Create the cekit image descriptor for Op…

### PR DESCRIPTION
**Pull request by davidesalerno**
…erator bundle image

**Description of the change:**

**Motivation for the change:**

**Checklist**

- [ ] Add or Modify a unit test for your change
- [ ] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: config/manifests/osl/bases/logic-operator-rhel8.clusterserviceversion.yaml
Comment (rgdoliveira) [inline config/manifests/osl/bases/logic-operator-rhel8.clusterserviceversion.yaml:60]: You explained to me that this is just for the upgrade. In that case, can't we use the tag `1.29.0` instead of the `sha256`? Or the `sha256` is really needed?

I'm just thinking that it would be easier if we don't have to update this field with the sha of the Operator image and just use the tag.

Comment (spolti) [reply config/manifests/osl/bases/logic-operator-rhel8.clusterserviceversion.yaml:60]: IMHO for now we should stick with the tag.
For the sha256, it is better to let osbs handle it.

File: bundle/osl/manifests/logic-operator-rhel8.clusterserviceversion.yaml
Comment (spolti) [reply bundle/osl/manifests/logic-operator-rhel8.clusterserviceversion.yaml:441]: +1 on stick with 1.29

Later on if we decide to support disconnected installations, we might need to do it, but, in this case, osbs handle the sha for us.

File: Makefile.osl
Comment (spolti) [inline Makefile.osl:1]: There is a way to import another makefiles, this way we could import the other one and only update what is needed to build the product image, wdyt?

File: Makefile.osl
Comment (rgdoliveira) [reply Makefile.osl:168]: This is the cekit `--release` flag when building in OSBS. If you don't use it, you will only have scratch builds.

So yeah, If we want to have a non-scratch build on OSBS, this is needed.

File: bundle/osl/manifests/logic-operator-rhel8.clusterserviceversion.yaml
Comment (rgdoliveira) [inline bundle/osl/manifests/logic-operator-rhel8.clusterserviceversion.yaml:527]: @davidesalerno Shouldn't we change the Operator to use the tag `1.29.0` instead of the sha256 as as you did for file `config/manifests/osl/bases/logic-operator-rhel8.clusterserviceversion.yaml` ?

File: api/v1alpha08/sonataflow_types.go
Comment (MarianMacik) [inline api/v1alpha08/sonataflow_types.go:202]: Maybe also ConfigMap for devmode?

File: api/v1alpha08/sonataflowplatform_types.go
Comment (MarianMacik) [inline api/v1alpha08/sonataflowplatform_types.go:225]:
```suggestion
// +operator-sdk:csv:customresourcedefinitions:resources={{Namespace,v1,"The Namespace controlled by the platform"}}
```

File: api/v1alpha08/sonataflowbuild_types.go
Comment (MarianMacik) [inline api/v1alpha08/sonataflowbuild_types.go:110]:
```suggestion
// +operator-sdk:csv:customresourcedefinitions:resources={{BuildConfig,build.openshift.io/v1,"An Openshift Build Config"}}
```

File: Makefile.osl
Comment (davidesalerno) [reply Makefile.osl:1]: Using 
`.INCLUDE : Makefile`
we can import the variables but it seems that it doesn't import the targets so we have to put something like 
`@$(MAKE) -f Makefile $@`
for every unchanged target.
Do yo know a better way?

File: api/v1alpha08/sonataflow_types.go
Comment (davidesalerno) [reply api/v1alpha08/sonataflow_types.go:202]: I added the missing API Marker for ConfigMap ;-)

File: bundle/app/manifests/sonataflow-operator.clusterserviceversion.yaml
Comment (spolti) [inline bundle/app/manifests/sonataflow-operator.clusterserviceversion.yaml:539]: isn't is supposed to be Red Hat?

**Review by spolti — APPROVED** [review]

File: bundle/app/manifests/sonataflow-operator.clusterserviceversion.yaml
Comment (davidesalerno) [reply bundle/app/manifests/sonataflow-operator.clusterserviceversion.yaml:539]: Probably for the community version it won't be Red Hat but Apache.
@ricardozanini can you confirm?

Comment (ricardozanini) [reply bundle/app/manifests/sonataflow-operator.clusterserviceversion.yaml:539]: Provider should be Red Hat, we change to Apache once we do the move.

Comment (davidesalerno) [reply bundle/app/manifests/sonataflow-operator.clusterserviceversion.yaml:539]: I reverted it to Red Hat

**Review by desmax74 — APPROVED** [review]

File: bundle.Dockerfile
Comment (ricardozanini) [inline bundle.Dockerfile:18]: Why this change? Won't it break the operator-sdk scaffold compatibility?

File: bundle/app/manifests/sonataflow-operator.clusterserviceversion.yaml
Comment (ricardozanini) [inline bundle/app/manifests/sonataflow-operator.clusterserviceversion.yaml:208]: You're breaking the description.

File: bundle/app/manifests/sonataflow-operator.clusterserviceversion.yaml
Comment (ricardozanini) [inline bundle/app/manifests/sonataflow-operator.clusterserviceversion.yaml:98]: Please revert these annotations.

File: docs/CONTRIBUTING.md
Comment (ricardozanini) [inline docs/CONTRIBUTING.md:155]: I'd rather not do this. If you want, we can discuss if it would be better to create another branch/repo for productization.

Doing so will break operator-sdk/kubebuilder scaffolding.

**Review by ricardozanini — CHANGES_REQUESTED** [review]
Let's review this approach. It's too intrusive IMO. We can have a new repo just with prod bits if you want to use this as an upstream. Similar to what OCP/Knative does.

File: bundle.Dockerfile
Comment (davidesalerno) [reply bundle.Dockerfile:18]: No because we updated the Makefile.

If we would like to have SonataFlow reference in the community CSV and OpenShift Serverless Logic into the productized CSV it is the only way to go.

File: bundle/app/manifests/sonataflow-operator.clusterserviceversion.yaml
Comment (davidesalerno) [reply bundle/app/manifests/sonataflow-operator.clusterserviceversion.yaml:98]: Ok, unfortunately [this](https://github.com/kiegroup/kogito-serverless-operator/commit/ab7f9cb61c537cc0a0b9ebd24c168d70db5f1a9a) was merged after the rebase with main and I've lost that changes

File: docs/CONTRIBUTING.md
Comment (davidesalerno) [reply docs/CONTRIBUTING.md:155]: This is not breaking anything if managed like we are doing.

Having another branch only for the productization will add a lot of effort due to the branch alignments and will prevent us to have smart CI checks that could let us know if the code is everytime ready to be productized.

Anyway we can discuss about the 2 approaches.

**Review by ricardozanini — COMMENTED** [review]
Waiting for the reverse of moving to `app` dir.

File: bundle.osl/manifests/logic-operator-rhel8-controller-manager-metrics-service_v1_service.yaml
Comment (davidesalerno) [reply bundle.osl/manifests/logic-operator-rhel8-controller-manager-metrics-service_v1_service.yaml:2]: This is the output produced by the operator-sdk command `operator-sdk generate bundle` based on what it is into the config directory.

Are you meaning that we can create there a base kustomization?

Comment (ricardozanini) [reply bundle.osl/manifests/logic-operator-rhel8-controller-manager-metrics-service_v1_service.yaml:2]: Yes, instead of duplicating the bundle dir like you're doing, we can use a base kustomize to just change the naming to the "productized" one.

File: bundle.osl/manifests/logic-operator-rhel8-builder-config_v1_configmap.yaml
Comment (ricardozanini) [reply bundle.osl/manifests/logic-operator-rhel8-builder-config_v1_configmap.yaml:5]: Also, this is something that kustomize can do.

File: bundle.osl/manifests/logic-operator-rhel8-controller-manager-metrics-service_v1_service.yaml
Comment (davidesalerno) [reply bundle.osl/manifests/logic-operator-rhel8-controller-manager-metrics-service_v1_service.yaml:2]: I tried to simplify the structure of the config folder limiting the duplication but maintaining the "compatibility" with the operator-sdk with these changes https://github.com/kiegroup/kogito-serverless-operator/pull/145/commits/057b3f0848e6ace5d59a9ac653006b599c6a87cb

In order to increase the level of parametrization into the kunstomize files we should got for the bases and overlay approach https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization/#bases-and-overlays but we will loose the compatibility with the operator-sdk.

File: config/manager/manager.yaml
Comment (ricardozanini) [inline config/manager/manager.yaml:30]: Why are we removing this?

File: osl-operator.yaml
Comment (ricardozanini) [inline osl-operator.yaml:1]: You can remove this file, the prod one should only be installed via OLM.

File: bundle.Dockerfile
Comment (davidesalerno) [reply bundle.Dockerfile]: I restore the default folder structure for the community release

File: config/manager/manager.yaml
Comment (davidesalerno) [reply config/manager/manager.yaml:30]: Because it is not needed and in fact it is blocking the CVP tests to pass

File: osl-operator.yaml
Comment (ricardozanini) [reply osl-operator.yaml:1]: I think they should install a subscription with the custom image or run the operator locally interacting with the cluster directly.

Comment (davidesalerno) [reply osl-operator.yaml]: I removed that file

File: bundle.osl/manifests/logic-operator-rhel8-builder-config_v1_configmap.yaml
Comment (ricardozanini) [reply bundle.osl/manifests/logic-operator-rhel8-builder-config_v1_configmap.yaml:5]: Aren't you changing this?

**Review by ricardozanini — COMMENTED** [review]
I asked you offline, but I'll let it registered here. Are we generating the osl files or they are just copied and pasted? Even if we can't use kustomize for everything, maybe we need a script that would create these files based upstream, so we avoid having to remember to change twice something. For example, the configMap or the manager deployment.

File: bundle.osl/manifests/logic-operator-rhel8-builder-config_v1_configmap.yaml
Comment (davidesalerno) [reply bundle.osl/manifests/logic-operator-rhel8-builder-config_v1_configmap.yaml]: Updated with the productized version

**Review by ricardozanini — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

File: bundle.osl/manifests/logic-operator-rhel8-builder-config_v1_configmap.yaml
Comment (radtriste) [inline bundle.osl/manifests/logic-operator-rhel8-builder-config_v1_configmap.yaml:5]: Isn't that dangerous to use `latest` ? I would set it to current prod version, aka `1.30`

File: config/default/osl/manager_auth_proxy_patch.yaml
Comment (radtriste) [inline config/default/osl/manager_auth_proxy_patch.yaml:13]: Having a sha is more difficult to follow-up on which version we are using. And does not update in case of bug fix releases

File: config/manager/osl/sonataflow_builder_dockerfile.yaml
Comment (radtriste) [inline config/manager/osl/sonataflow_builder_dockerfile.yaml:1]: we should be using a defined version, aka `1.30`

File: config/manifests/bases/sonataflow-operator.clusterserviceversion.yaml
Comment (radtriste) [inline config/manifests/bases/sonataflow-operator.clusterserviceversion.yaml:8]: why this set back ?

File: Makefile.osl
Comment (radtriste) [inline Makefile.osl:6]: can you sync with the `Makefile`. We introduced a new `REDUCED_VERSION` in there

Why not use the `include Makefile` and redefine the needed targets only ?

File: hack/align-osl-config.sh
Comment (davidesalerno) [reply hack/align-osl-config.sh:22]: Unfortunately there are several folders with the `osl` subdirs into the `config` directory but only this one need to be automatically aligned.

File: config/manager/osl/sonataflow_builder_dockerfile.yaml
Comment (davidesalerno) [reply config/manager/osl/sonataflow_builder_dockerfile.yaml:1]: Yes sure, is this needed also for the 'main'?

File: Makefile.osl
Comment (davidesalerno) [reply Makefile.osl:6]: Unfortunately I noticed that the `include Makefile` works for targets but not for variables and so even if we can override the targets there is an issue with the variables.

File: config/manifests/bases/sonataflow-operator.clusterserviceversion.yaml
Comment (davidesalerno) [reply config/manifests/bases/sonataflow-operator.clusterserviceversion.yaml]: Reverted

File: content_sets.yaml
Comment (davidesalerno) [reply content_sets.yaml]: Updated

File: hack/align-osl-config.sh
Comment (davidesalerno) [reply hack/align-osl-config.sh:1]: I added it

**Comment by mareknovotny** [comment]
what is holding this PR from merging? Is there any remaining work here? Do you know @ricardozanini ?

**Comment by ricardozanini** [comment]
@mareknovotny there are a few questions unanswered by Davide. I'll take this work since it looks like we don't have anyone else to do this work. But on my timeframe.

**Comment by mareknovotny** [comment]
i guess @davidesalerno can answer them if they are needed ;)

**Comment by spolti** [comment]
> @mareknovotny there are a few questions unanswered by Davide. I'll take this work since it looks like we don't have anyone else to do this work. But on my timeframe.

I think we can merge this one as address any minor issue later.
As it refers to prod builds only, it would not affect the current releases nor introduce bugs.

File: config/default/osl/manager_auth_proxy_patch.yaml
Comment (davidesalerno) [reply config/default/osl/manager_auth_proxy_patch.yaml:13]: For productization is the approach we are following as suggested by the OpenShift Serverless team

File: bundle.osl/manifests/logic-operator-rhel8-builder-config_v1_configmap.yaml
Comment (davidesalerno) [reply bundle.osl/manifests/logic-operator-rhel8-builder-config_v1_configmap.yaml:5]: Since we are on main as discussed we could use this and switch on each version branch to the right version

**Comment by davidesalerno** [comment]
@ricardozanini I reported here the answers we agreed on in our pasts weekly alignments.
I think that as soon as the CVP tests are fine we should merge this PR and eventually open some improvement tasks to tweak some points (like the Makefile, if it is possible) because doing everything in one shot it is blocking this change and it is taking a lot of efforts to regaling it to the latest main version every time.

**Comment by ricardozanini** [comment]
@davidesalerno sure thing, just let me know when it's ready.

**Pull request merged**
